Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderTableView.cpp

```diff
@@ -69,8 +71,23 @@ namespace AzQtComponents
 
     void AssetFolderTableView::selectionChanged(const QItemSelection& selected, const QItemSelection& deselected)
     {
+        m_selectionChangedSinceClick = true;
         TableView::selectionChanged(selected, deselected);
         Q_EMIT selectionChangedSignal(selected, deselected);
+
+    }
+
+    void AssetFolderTableView::onClickedView([[maybe_unused]] const QModelIndex& index)
+    {
+        // if we click on an item and selection wasn't changed, then reselect the current item so that it shows up in
+        // any related previewers.
+        bool didSelectionChange = m_selectionChangedSinceClick;
```

Comment:
I'm not a fan of keeping this state around via a bool as it might get outdated with future changes. Looking at the code seems like your use case is to know if the item you just clicked on was already selected. So I would suggest to grab the selection model from the tableview, get the data from the index, and see if your item is in there already. If the selection model is already updated with your selection when your reach this callback, then there is surely a widget or some state somewhere already that you can check to see if it already uses your item

Reply:
the use case is to click on an item that was already selected, yes.

I'll give it a try though, its basically one of two situations

Either this function is called BEFORE the selection model is updated, in which case, its an easy check, as you say, or, its called AFTER the selection model already updates, in which case instead of caching a bool, we'd have to cache the prior selection, which is a lot more troublesome.

Reply:
I was able to get it to work, meaning that it seems to work just fine if we just check the model without any extra state. Updated